File: Matlab/sym_to_data_ang.m
function [data] = sym_to_data_ang(symbol,N)
%SYM_TO_DATA returns an N-sample upchirp of data symbol

    data = [];
    accumulator = 0;
    
    for j = symbol
        phase = -pi + ((j-1)*(2*pi/(N)));
        temp = [];

        for i = 1:N
            accumulator = accumulator + phase;
            polar_radius = 1;

            [x, y] = pol2cart(accumulator, polar_radius);

            temp(i) = complex(x, y);

            phase = phase + (2*pi/(N));
        end
        data = [data temp];
    end
end

File: Python/sym_to_data_ang.py
import numpy as np
import math
from util import pol2cart

def sym_to_data_ang(symbols,N):
    data = []
    accumulator = 0
    
    for j in symbols:
        phase = -math.pi + ((j-1)*(2*math.pi/(N)))
        temp = []
        for i in range(N):
            accumulator = accumulator + phase
            polar_radius = 1
            [x, y] = pol2cart(accumulator, polar_radius)
            temp.append(x + 1j*y)
            phase = phase + (2*math.pi/(N))
        data += temp
    return np.array(data)


